pkg/didcomm/common/service/did_comm_msg.go

```diff
@@ -25,6 +25,19 @@ const (
 	jsonThreadID = "thid"
 )
 
+// MetaData may contain additional payload for the protocol. It might be populated by the client/protocol
+// for outbound messages. If metadata were populated, the messenger will automatically add it to the incoming
+// messages by the threadID. If Metadata is <nil> in the outbound message the previous payload
+// will be added to the incoming message. Otherwise, the payload will be rewritten.
+// NOTE: Metadata is not a part of the JSON message. The payload will not be sent to another agent.
+// Metadata should be used by embedding it to the model structure. e.g
+// 	type A struct {
+// 		MetaData `json:",squash"`
+// 	}
+type MetaData struct {
```

Comment:
I assume this should be `Metadata`?

Reply:
If you have some suggestions about naming, let me know! The problem is type and field name should be different.
```
type MetaData struct {
	Metadata map[string]interface{} `json:"metadata,omitempty"`
}
```

Reply:
Seems like a structure related to a Msg or Messenger?

Examples (there are probably better ideas):
- MsgInfo
- MessengerData

Reply:
Actually, the messenger doesn't need this. It is a protocol-specific payload. For me, metadata is the best name.  I've modified it. Also, I have added an accesor for it.

```
type Metadata struct {
	Payload map[string]interface{} `json:"metadata,omitempty"`
}

```